src/storage/data_table.cpp

```diff
@@ -125,31 +127,306 @@ DataTable::~DataTable() {
+      // TODO: throw an exception
+      LOG_ERROR("Operator NOT SUPPORTED");
+      return false;
+    }
+  }
+  return true;
+} */
+
+// Set the default values for corresponding columns
+/*
+bool DataTable::SetDefaults(storage::Tuple *tuple) {
   PL_ASSERT(schema->GetColumnCount() == tuple->GetColumnCount());
 
   oid_t column_count = schema->GetColumnCount();
   for (oid_t column_itr = 0; column_itr < column_count; column_itr++) {
-    if (tuple->IsNull(column_itr) && schema->AllowNull(column_itr) == false) {
-      LOG_TRACE(
-          "%u th attribute in the tuple was NULL. It is non-nullable "
-          "attribute.",
-          column_itr);
-      return false;
+    if (tuple->IsNull(column_itr) && schema->AllowDefault(column_itr)) {
+      auto default_value = schema->GetDefaultValue(column_itr);
+      tuple->SetValue(column_itr, *default_value);
+      return true;
     }
   }
 
   return true;
+} */
+
+bool DataTable::CheckExp(const storage::Tuple *tuple, oid_t column_idx,
+                         std::pair<ExpressionType, type::Value> exp) const {
+  type::Value cur = tuple->GetValue(column_idx);
+  switch (exp.first) {
+    case ExpressionType::COMPARE_EQUAL: {
+      if (cur.CompareNotEquals(exp.second) == type::CMP_TRUE) return false;
+      break;
+    }
+    case ExpressionType::COMPARE_NOTEQUAL: {
+      if (cur.CompareEquals(exp.second) == type::CMP_TRUE) return false;
+      break;
+    }
+    case ExpressionType::COMPARE_LESSTHAN: {
+      if (cur.CompareGreaterThanEquals(exp.second) == type::CMP_TRUE)
+        return false;
+      break;
+    }
+    case ExpressionType::COMPARE_GREATERTHAN: {
+      if (cur.CompareLessThanEquals(exp.second) == type::CMP_TRUE) return false;
+      break;
+    }
+    case ExpressionType::COMPARE_LESSTHANOREQUALTO: {
+      if (cur.CompareGreaterThan(exp.second) == type::CMP_TRUE) return false;
+      break;
+    }
+    case ExpressionType::COMPARE_GREATERTHANOREQUALTO: {
+      if (cur.CompareLessThan(exp.second) == type::CMP_TRUE) return false;
+      break;
+    }
+    default: {
+      // TODO: throw an exception
+      LOG_ERROR("Operator NOT SUPPORTED");
+      return false;
+    }
+  }
+  return true;
 }
 
 bool DataTable::CheckConstraints(const storage::Tuple *tuple) const {
   // First, check NULL constraints
-  if (CheckNulls(tuple) == false) {
-    LOG_TRACE("Not NULL constraint violated");
-    throw ConstraintException("Not NULL constraint violated : " +
-                              std::string(tuple->GetInfo()));
-    return false;
+  PL_ASSERT(schema->GetColumnCount() == tuple->GetColumnCount());
+  oid_t column_count = schema->GetColumnCount();
+  for (oid_t column_itr = 0; column_itr < column_count; column_itr++) {
+    std::vector<catalog::Constraint> column_cons =
+        schema->GetColumn(column_itr).GetConstraints();
+    for (auto cons : column_cons) {
+      ConstraintType type = cons.GetType();
+      switch (type) {
+        case ConstraintType::NOTNULL:
```

Comment:
What is the difference between `ConstraintType::NOTNULL` and `ConstraintType::NOT_NULL` ?